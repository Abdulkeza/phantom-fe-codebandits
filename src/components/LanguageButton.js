import React from 'react';

const LanguageButton = () => {
  return (
    <div>
      <select className="bg-background border outline-none md:ml-6 lg:ml-6 xl:ml-6 mt-5 flex py-2 px-4 text-sm hover:font-bold">
        <option className=" border-none bg-background w-full cursor-pointer">
          English
        </option>
        <option className=" border-none bg-background w-full cursor-pointer">
          French
        </option>
        <option className=" border-none bg-background cursor-pointer">
          Kinyarwanda
        </option>
      </select>
    </div>
  );
};

export default LanguageButton;
