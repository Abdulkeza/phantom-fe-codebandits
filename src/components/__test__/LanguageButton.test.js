import React from 'react';
import renderer from 'react-test-renderer';
import { mount } from 'enzyme';
import { MemoryRouter } from 'react-router-dom';
import LanguageButton from '../LanguageButton.js';

describe('LanguageButton', () => {
  it('should render LanguageButton', () => {
    const elem = renderer.create(<LanguageButton />).toJSON();
    expect(elem).toMatchSnapshot();
  });
  it('It should test the setOpen function', () => {
    const setOpen = jest.fn();
    const wrapper = mount(
      <MemoryRouter>
        <LanguageButton setOpen={setOpen()} />
      </MemoryRouter>
    );
    const button = wrapper.find('button');
    button.simulate('click');
    expect(setOpen).toBeCalledTimes(1);
  });
});
