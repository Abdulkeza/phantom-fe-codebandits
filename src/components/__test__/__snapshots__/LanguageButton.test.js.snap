// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LanguageButton should render LanguageButton 1`] = `
<div>
  <button
    className="undefined md:mt-8 lg:mt-8 xl:mt-8 flex py-2 px-4 text-lg hover:font-bold"
    onClick={[Function]}
    type="button"
  >
    <h1>
      English
    </h1>
    <img
      alt="dropdown"
      className="w-3 mt-3 ml-2 h-2"
      src="dropdown.png"
    />
  </button>
</div>
`;
